frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelectionPopupTestPartialFailed.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { render, waitFor } from '@testing-library/react';
+import { setupStore } from '@test/utils/setupStoreUtil';
+import { Provider } from 'react-redux';
+
+import { PipelineMetricSelection } from '@src/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection';
+import { IPipelineConfig, updateShouldGetPipelineConfig } from '@src/context/Metrics/metricsSlice';
+import { addNotification } from '@src/context/notification/NotificationSlice';
+import { METRICS_DATA_FAIL_STATUS } from '@src/constants/commons';
+import { PIPELINE_SETTING_TYPES } from '@test/fixtures';
+
+const store = setupStore();
+let stepFailStatus = METRICS_DATA_FAIL_STATUS.NOT_FAILED;
+
+jest.mock('@src/context/notification/NotificationSlice', () => ({
+  ...jest.requireActual('@src/context/notification/NotificationSlice'),
+  addNotification: jest.fn().mockReturnValue({ type: 'ADD_NEW_NOTIFICATION' }),
```

Comment:
Since in line-7 `addNotification` is already imported, why mock it here,
Did I missed any further consideration?

BTW, I think the notification behavior could be assertted via some UI presentation instead of asserting function invoiking. Unless asserting invoking is the only way.

Reply:
Since we are expecting the "addNotification" to be called in the unit tests below, we need to mock it in the first place, otherwise, error will occur.